### PR TITLE
docs: Update Appium documentation

### LINKED ISSUE #11123: Docs: Update Appium section in readme
### What is this about?

This task aims to enhance the Appium section's clarity, ensuring that everyone understands how tests are executed and triggered. The section needs refinement to provide more straightforward step-by-step instructions so the process is easy to follow and all team members are aligned.

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

A clear and concise appium test section is created in the tests readme. 

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by cortisiko**
## **Description**

The goal of this task is to enhance the clarity of the Appium section, ensuring that everyone understands how tests are executed and triggered. The section needs refinement to provide clearer step-by-step instructions, so the process is easy to follow and all team members are aligned.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by Cal-L — DISMISSED** [review]
LGTM

File: wdio/config/ios.config.browserstack.js [hunk 1]
```diff
@@ -1,24 +1,64 @@
+import { removeSync } from 'fs-extra';
```
Comment (cortisiko) [inline wdio/config/ios.config.browserstack.js:1]: There were redundant browserstack configs. One for BrowserStack Local and a Browserstack "prod" config. There is no difference between the two configs; both run the tests against browserstack, and you can use either to run tests locally or on CI.

**Review by SamuelSalas — DISMISSED** [review]

**Review by chrisleewilcox — APPROVED** [review]
LGTM

**Comment by NicolasMassart** [comment]
A but late to comment as it's merged already, but I wanted to say it's nice work!

**Pull request merged**